locator/tablets.cc

```diff
@@ -263,9 +263,29 @@ class tablet_effective_replication_map : public effective_replication_map {
     virtual inet_address_vector_replica_set get_natural_endpoints(const token& search_token) const override {
         auto&& tablets = get_tablet_map();
         auto tablet = tablets.get_tablet_id(search_token);
-        auto&& replicas = tablets.get_tablet_info(tablet).replicas;
-        tablet_logger.trace("get_natural_endpoints({}): table={}, tablet={}, replicas={}", search_token, _table, tablet, replicas);
-        return to_replica_set(replicas);
+        auto* info = tablets.get_tablet_transition_info(tablet);
+        if (!info) {
+            auto&& replicas = tablets.get_tablet_info(tablet).replicas;
+            tablet_logger.trace("get_natural_endpoints({}): table={}, tablet={}, replicas={}", search_token, _table, tablet, replicas);
+            return to_replica_set(replicas);
+        }
+        switch (info->stage) {
+            case tablet_transition_stage::allow_write_both_read_old:
+            case tablet_transition_stage::write_both_read_old:
+            case tablet_transition_stage::write_both_read_new:
+            case tablet_transition_stage::streaming: {
+                auto&& replicas = tablets.get_tablet_info(tablet).replicas;
+                tablet_logger.trace("get_natural_endpoints({}): table={}, tablet={}, replicas={}", search_token, _table, tablet, replicas);
+                return to_replica_set(replicas);
+            }
+            case tablet_transition_stage::use_new:
+            case tablet_transition_stage::cleanup: {
+                auto&& replicas = info->next;
+                tablet_logger.trace("get_natural_endpoints({}): table={}, tablet={}, replicas={}", search_token, _table, tablet, replicas);
+                return to_replica_set(replicas);
+            }
+        }
+        abort();
```

Comment:
These should be on_internal_error()s.

Reply:
Even though the compiler guarantees that we can't reach this place (there is no default case)?

Reply:
The compiler only guarantees we don't forget an enum. We can still have enums with bad values.

Also, it's a way to document that this case as considered (on_internal_error vs. on_fatal_internal_error).